arroyo/types.py

```diff
@@ -63,3 +63,9 @@ def __repr__(self) -> str:
         # ``__slots__`` for performance reasons. The class variable names
         # would conflict with the instance slot names, causing an error.
         return f"{type(self).__name__}(partition={self.partition!r}, offset={self.offset!r})"
+
+
+@dataclass
+class Offset:
+    offset: int
```

Comment:
Nit: Reading code like offset.offset is harder to read. If you rename this to kafka_offset, reading code like offset.kafka_offset would be much more explicit

Reply:
Thanks, updated to kafka_offset. Naming is hard.

Reply:
I don't agree with this. This was intentionally written to be independent from Kafka so that it other conceptually similar backend implementations ([local backend for testing](https://github.com/getsentry/arroyo/blob/main/arroyo/backends/local/backend.py), Pulsar, Google PubSub, Redis Streams, PostgreSQL logical slots, etc.) could be substituted with minimal effort and ideally used by other teams in different environments. Referencing Kafka by name here clearly breaks that encapsulation.

"Offset" may have been too loaded of a term to begin with in the original method signatures (for the same reason — it is too implicitly coupled to a specific backend implementation.) Maybe one of "Cursor" or "Position" would have been more appropriate in the first place. I think `offset.offset` is more of an indication that the conceptual model has changed so that this no longer represents a simple offset but something different.

Aside, considering this data structure is seemingly immutable, it should probably be a `NamedTuple` for efficiency and immutability (or a dataclass with `__slots__` defined and `frozen=True`.)

Reply:
Good point, `Position` makes sense to me as a container holding the offset and timestamp

Reply:
Seems like `commit_offsets`, `state_offsets`, etc. should also be updated for consistency with the conceptual change.

Reply:
Updated to stage_positions, commit_positions

Reply:
Thanks